### ISSUE TITLE
Evaluation through 

### ISSUE THREAD

**Issue by Arnav0400** [Lightning-AI/lit-llama#368]
Hello Team, Great work! I wanted to know whether https://github.com/Lightning-AI/lm-evaluation-harness/tree/lit-llama works or not since it is recently updated. If not is there any way I can evaluate lit-llama models on the[ HF-leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard) benchmark?

Thanks in advance @lantiga

**Comment by lantiga** [comment]
Hey @Arnav0400, it does work at least for HellaSwag and PiQA, which are the tests I focused on a few weeks ago
https://github.com/EleutherAI/lm-evaluation-harness/issues/539
I'm now also adding lit-parrot support as well.

It's a very sensible thing to make sure it works with all the evaluations required by HF-leaderboard, that is:
- AI2 Reasoning Challenge
- HellaSwag (already supported)
- MMLU
- TruthfulQA

Would you like to give it a shot and report back?

**Comment by lantiga** [comment]
I gave it a shot and the benchmarks seem to run on lit-llama. I'll publish an howto, but for the time being, this is the end-to-end rundown (I'm downloading OpenLLaMA weights here just as an example):

```
git clone git@github.com:Lightning-AI/lit-llama.git
cd lit-llama
pip install -e .

python scripts/download.py --repo_id openlm-research/open_llama_7b_700bt_preview --local_dir checkpoints/open-llama/7B
python scripts/convert_hf_checkpoint.py --checkpoint_dir checkpoints/open-llama/7B --model_size 7B

cd ..

git clone git@github.com:Lightning-AI/lm-evaluation-harness.git
cd lm-evaluation-harness
git checkout lit-llama
pip install -e .

# may be better to do
ln -s ../lit-llama/checkpoints checkpoints

python main.py --model lit-llama --model_args dtype=bfloat16 --tasks arc_challenge,hellaswag,hendrycks,truthfulqa_mc --device cuda:0 --write_out
```

**Comment by lantiga** [comment]
So what will unblock that specifically is a conversion script of checkpoints from lit-* to equivalent HF models, so you can upload on the hub.

We’ll try to schedule this shortly, but help will be appreciated!

**Comment by Arnav0400** [comment]
Yes, but it should be pretty simpler I guess, I just have one concern - the time taken for evaluation seems to be higher than other huggingface models. Can you confirm this? I can maybe look at how to make it efficient then.

**Comment by lantiga** [comment]
It's quite possible because I'm only supporting `batch_size` of 1 atm (I was only interested in the numbers at the time so that was enough for me)

**Comment by Arnav0400** [comment]
Any headsup on how I can implement higher batch sizes? I will try to implement it over the weekend!
